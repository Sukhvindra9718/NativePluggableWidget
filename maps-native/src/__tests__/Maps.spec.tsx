/**
 * @jest-environment jsdom
 */
import { Maps, Props } from "../Maps";
import { mount } from "enzyme";
import { dynamicValue } from "@mendix/piw-utils-internal";
import { Big } from "big.js";
import { createElement } from "react";
import { waitFor } from "@testing-library/react-native";

describe("", () => {
    let defaultProps: Props;

    beforeEach(() => {
        defaultProps = {
            name: "maps-test",
            style: [],
            markers: [],
            dynamicMarkers: [],
            fitToMarkers: true,
            defaultZoomLevel: "city",
            minZoomLevel: "city",
            maxZoomLevel: "streets",
            mapType: "standard",
            provider: "default",
            interactive: true,
            showsUserLocation: false
        };
    });

    it("renders", async () => {
        defaultProps.markers = [
            {
                locationType: "latlng", // Tokyo
                latitude: dynamicValue(new Big(35.6895), false),
                longitude: dynamicValue(new Big(139.6917), false),
                iconSize: 1,
                iconColor: "red"
            },
            {
                locationType: "latlng", // Amsterdam
                latitude: dynamicValue(new Big(52.3702), false),
                longitude: dynamicValue(new Big(4.8952), false),
                iconSize: 1,
                iconColor: "red"
            }
        ];

        const wrapper = mount(<Maps {...defaultProps} />);

        await waitFor(() => {
            wrapper.update();
            expect(wrapper).toMatchSnapshot();
        });
    });
});
