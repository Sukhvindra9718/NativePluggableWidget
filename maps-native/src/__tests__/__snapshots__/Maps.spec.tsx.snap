// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<Maps
  defaultZoomLevel="city"
  dynamicMarkers={[]}
  fitToMarkers={true}
  interactive={true}
  mapType="standard"
  markers={
    [
      {
        "iconColor": "red",
        "iconSize": 1,
        "latitude": {
          "status": "available",
          "value": "35.6895",
        },
        "locationType": "latlng",
        "longitude": {
          "status": "available",
          "value": "139.6917",
        },
      },
      {
        "iconColor": "red",
        "iconSize": 1,
        "latitude": {
          "status": "available",
          "value": "52.3702",
        },
        "locationType": "latlng",
        "longitude": {
          "status": "available",
          "value": "4.8952",
        },
      },
    ]
  }
  maxZoomLevel="streets"
  minZoomLevel="city"
  name="maps-test"
  provider="default"
  showsUserLocation={false}
  style={[]}
>
  <View
    style={
      {
        "minHeight": 50,
      }
    }
    testID="maps-test"
  >
    <View
      style={
        {
          "minHeight": 50,
        }
      }
      testID="maps-test"
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#fafafa",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 99,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#fafafa",
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 99,
            }
          }
        >
          <ActivityIndicator
            size="large"
          >
            <ActivityIndicator
              size="large"
            />
          </ActivityIndicator>
        </View>
      </View>
    </View>
  </View>
</Maps>
`;
